src/secrets/CommandRemove.ts

```diff
@@ -59,25 +52,60 @@ class CommandRemove extends CommandPolykey {
           options: { nodePath: options.nodePath },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        await binUtils.retryAuthentication(async (auth) => {
+        const hasErrored = await binUtils.retryAuthentication(async (auth) => {
           const response =
             await pkClient.rpcClient.methods.vaultsSecretsRemove();
           const writer = response.writable.getWriter();
           let first = true;
-          for (const [vault, path] of secretPaths) {
+          for (const [vaultName, secretPath] of secretPaths) {
             await writer.write({
-              nameOrId: vault,
-              secretName: path,
+              nameOrId: vaultName,
+              secretName: secretPath ?? '/',
               metadata: first
                 ? { ...auth, options: { recursive: options.recursive } }
                 : undefined,
             });
             first = false;
           }
           await writer.close();
-          // Wait for the program to generate a response (complete processing).
-          await response.output;
+          // Check if any errors were raised
+          let hasErrored = false;
+          for await (const chunk of response.readable) {
+            if (chunk.type === 'error') {
+              hasErrored = true;
+              // TypeScript cannot properly decode this type, so we need to use
+              // the `as` keyword so it doesn't throw errors.
+              const error = chunk as ErrorMessage;
+              if (error.code === 'ENOTEMPTY') {
+                // Attempt to delete the directory without recursive set
+                process.stderr.write(
+                  `rm: cannot remove '${error.reason}': Is a directory\n`,
+                );
+                continue;
+              } else if (error.code === 'ENOENT') {
+                // Attempt to delete a non-existent path
+                process.stderr.write(
+                  `rm: cannot remove '${error.reason}': No such file or directory\n`,
+                );
+                continue;
+              } else if (error.code === 'EINVAL') {
+                // Attempt to delete vault root
+                process.stderr.write(
+                  `rm: cannot remove '${error.reason}': Permission denied\n`,
+                );
+              } else {
```

Comment:
When attempting to remove the vault root directory, the `VaultOps` operation throws an `EncryptedEFS` error with the code `'EINVAL'` for an invalid operation. For now, I have just captured this invalid operation as an output and am rendering it to the user as if they were trying to operate on a write-protected directory.